flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/plan/util/FlinkNodeOptUtil.scala

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.flink.table.plan.util
+
+import org.apache.flink.table.plan.nodes.calcite.Sink
+import org.apache.flink.table.plan.nodes.exec.{ExecNode, ExecNodeVisitorImpl}
+
+import com.google.common.collect.{Maps, Sets}
+import org.apache.calcite.rel.RelNode
+import org.apache.calcite.sql.SqlExplainLevel
+
+import java.io.{PrintWriter, StringWriter}
+import java.util
+import java.util.concurrent.atomic.AtomicInteger
+
+object FlinkNodeOptUtil {
```

Comment:
We should stop adding utility class if we cannot see a clear responsibility through the class name.

Reply:
The responsibility of this utility class is converting an exec node plan to a string. I think ExecNodePlanDumper is better than FlinkNodeOptUtil.